### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by AntonMurashov**

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -16,16 +23,16 @@ describe('AppComponent', () => {
...
   });
-
+/*
   it('should render title', () => {
     const fixture = TestBed.createComponent(AppComponent);
     fixture.detectChanges();
     const compiled = fixture.debugElement.nativeElement;
     expect(compiled.querySelector('.content span').textContent).toContain('angular-app app is running!');
-  });
+  });*/
```
Comment (AinuraAliieva) [inline src/app/app.component.spec.ts]: looks like we don't need this test at all, so please remove it :)

File: src/app/app.component.ts [hunk 2]
```diff
@@ -6,5 +6,5 @@ import { Component } from '@angular/core';
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
-  
+  title = 'AngularApp';
```
Comment (AinuraAliieva) [inline src/app/app.component.ts]: could you please clarify why we need it?

File: src/app/core/header/header.component.spec.ts [hunk 3]
```diff
@@ -22,4 +23,22 @@ describe('HeaderComponent', () => {
...
+    let button = fixture.debugElement.nativeElement.querySelector("img[src='assets/img/ic-exit.svg']");
+    button.click();
+
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should log on exit', () => {
+    const spy = spyOn(console, 'log');
+
+    let button = fixture.debugElement.nativeElement.querySelector("img[src='assets/img/ic-exit.svg']");
```
Comment (AinuraAliieva) [inline src/app/core/header/header.component.spec.ts]: please use `By.css` offered by Angular - [link](https://angular.io/guide/testing#bycss)

for all querying

Comment (AinuraAliieva) [inline src/app/core/header/header.component.spec.ts]: Consider adding a specific class to `img` tag, in that case, we will not tie to the source link, so we will be more flexible.

File: src/app/courses/course-item/course-item.component.spec.ts [hunk 4]
```diff
@@ -1,6 +1,60 @@
...
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TestHostComponent);
+    testHost = fixture.componentInstance;
+    fixture.detectChanges();
+  });  
+
+  it('should emit delete item', () => {
+    fixture.detectChanges();
```
Comment (AinuraAliieva) [inline src/app/courses/course-item/course-item.component.spec.ts]: we detect changes before each test so this detection looks useless

File: src/app/app.component.spec.ts (hunk 1 above)
Comment (AntonMurashov) [reply src/app/app.component.spec.ts]: Oops, removed :)

File: src/app/app.component.ts (hunk 2 above)
Comment (AntonMurashov) [reply src/app/app.component.ts]: Tried some testing and forgot to remove - done, together with respective test

File: src/app/core/header/header.component.spec.ts (hunk 3 above)
Comment (AntonMurashov) [reply src/app/core/header/header.component.spec.ts]: Fixed

Comment (AntonMurashov) [reply src/app/core/header/header.component.spec.ts]: Done

File: src/app/courses/course-item/course-item.component.spec.ts (hunk 4 above)
Comment (AntonMurashov) [reply src/app/courses/course-item/course-item.component.spec.ts]: Removed this call and other useless ones

**Pull request merged**